src/org/pentaho/reporting/platform/plugin/output/CachingPageableHTMLOutput.java

```diff
@@ -171,9 +171,11 @@ public synchronized int generate( final MasterReport report, final int acceptedP
         logger.warn( "Using cached report data for " + key );
         final ReportProgressListener listener = ReportListenerThreadHolder.getListener();
         if ( listener != null ) {
-          listener.reportProcessingFinished(
+          final ReportProgressEvent event =
             new ReportProgressEvent( "Cache", ReportProgressEvent.GENERATING_CONTENT, 0, 0,
```

Comment:
Events in Java use the first parameter in their constructor to tell who is generating the event. So the string "Cache" is not the originator, your CachingHTMLReportOutput instance is. So please change that to "this".